components/status-message/src/StatusMessage.tsx

```diff
@@ -0,0 +1,47 @@
+import { FC, createElement as h } from 'react';
+import { StandardProps, classBuilder } from '@not-govuk/component-helpers';
+import { AnchorList } from "@not-govuk/anchor-list";
+
+import '../assets/StatusMessage.scss';
+
+export type StatusAction = {
+  /** Location to link to */
+  href: string
+  /** Text of the link */
+  text: string
+  /** Title of the link */
+  title?: string
+};
+
+export type StatusMessageProps = StandardProps & {
+  /** Summary of status */
+  status: string,
+  /** Extra information about the status */
+  children?: any,
+  /** Links to perform status related actions */
+  actions?: StatusAction[]
+};
+
+export const StatusMessage: FC<StatusMessageProps> = ({
+    classBlock,
+    classModifiers,
+    className,
+    status,
+    children,
+    actions,
+    ...attrs
+  }) => {
+  const classes = classBuilder('hods-status-message', classBlock, classModifiers, className);
+
+  return status || children ? (
```

Comment:
Now that `status` is mandatory you can drop the ternary.

Reply:
Well you say that, but it's not quite as "mandatory" as you might think. You can still call the component without any props and not get any compile error, so some defensive programming here is probably still useful... I can definitely lose the `children` though!

Reply:
Well Typescript should give you a build error.

And in JS if you break the contract you deserve everything you get.